rust/lance/src/dataset.rs

```diff
@@ -3039,6 +3089,53 @@ mod tests {
         assert!(fragments[0].metadata.deletion_file.is_some());
     }
 
+    #[rstest]
+    #[tokio::test]
+    async fn test_table_metadata(
```

Comment:
The main test I would like to see is that if you have 5 concurrent writers each adding a different metadata key, that each of them can succeed and their write is reflected in the final result. Same for deletion. That would let us know that conflict resolution is working correctly.

Reply:
I've updated `lance::dataset::transaction::tests::test_conflicts` to include the new operations. Is this sufficient?

I couldn't immediately find any other existing tests that use concurrent writers to base additional testing on. Do you know of any tests that use concurrent writers if`lance::dataset::transaction::tests::test_conflicts` is not sufficient? If no such tests exist yet, I would be tempted to add concurrent writer testing as a separate PR first before finishing off this one.

Reply:
You are testing the path where we *detect* conflicts, but not where we *resolve* them.

For example if you have two concurrent writers:

* Writer A: Set metadata x=1
* Writer B: Set metadata y=2

The final metadata should include **both** `x=1` and `y=2`. That's what I want tested.

Here is a test where we validate concurrent appends work right: https://github.com/lancedb/lance/blob/9c361feee9d12f72e5bca852ef0df31bbbb1807e/rust/lance/src/io/commit.rs#L838-L839

Reply:
Life has been a bit hectic lately so I've only just had a chance to look at this. I've added some initial tests [here](https://github.com/lancedb/lance/pull/2820/files#diff-b8249267a6671d3e0cfa81a9618e1d43d786eb825e32e37f275fbef9d780b4b4), noting that these tests will probably be updated once the other PR feedback has been incorporated.